test.py

```diff
@@ -365,14 +365,9 @@ class BoostTestSuite(UnitTestSuite):
 
     _exec_name_cache: Dict[str, str] = dict()
 
-    def _generate_cache(self) -> None:
-        # Apply combined test only for test/boost
-        exe_path = pathlib.Path(self.mode, "test", self.name, 'combined_tests')
-        if self.name != 'boost' or not exe_path.exists():
-            return
-        exe = path_to(self.mode, "test", self.name, 'combined_tests')
+    def _generate_cache(self, exec_path, exec_name) -> None:
```

Comment:
I don't understand why you needed both exec_path and exec_name. Isn't exec_name just the basename of exec_path?

Reply:
yes, i didn't want to use `os.path.basename(exec_path)` in this function, as it is a reverse function of `path_to()`. but forgot to refactor the caller further like:

```py
exec_name = 'combined_tests`
exec_path = path_to(self.mode, "test", self.name, exec_name)
if self.name == 'boost' and os.path.exists(exec_path):
    self._generate_cache(exec_path, exec_name)
```

i will send a follow-up change to implement the above change once this change lands.